### DISCUSSION TITLE
wdirvish, a better wdired for dirvish

### DISCUSSION THREAD

**Discussion by Schievel1** [alexluigit/dirvish#241] [Ideas]
Editing text in emacs is powerful. In fact it is so powerful, we try to translate editing other things into editing text. This is what we do with wdired for example. 

To be honest, for a users not knowing emacs wdired feels like magic. You can edit files by just editing text in emacs, how cool is that?! But wdired is only half way there. You can edit the files names and permissions in a dired buffer (i.e. a folder of the filesystem) but you can not move or copy files to other directories. 
This is my idea, how cool would it be if we had two dirvish buffers, you could delete a line in one buffer and paste it into the other. While also being able to navigate to another folder in each of these buffers. 

Wdired does its thing by comparing the names of the old files with the new ones when we do `wdired-finish-edit` then doing a mv operation on the file when the names are different. Similarly we could read the list of files from before when entering wdirvish, then when the user does `wdirvish-finish-edit` it compares the new file list with the old and so we could figure out if a file should be copied between the two buffers, or moved of deleted in one buffer. 

Some things that need to be considered:
- How one could implement the folder navigation. If a user deletes one line in one buffer then navigates to another to paste it there we need to keep track of all the folder states every time a buffer is opened I guess. 
- Is this the right place? I like dirvish because it has those window layout modes. When entering wdirvish we could force a two pane layout which would make the implementation easier I guess.

**Comment by lengani** [comment]
A better wdired would be cool indeed. I think moving files is already possible with wdired if you set some variables. One thing I'd like to see in wdirvish is the ability to create new files in the buffer. Not via the minibuffer but in the buffer itself. Similar to how it is done in oil.nvim (https://github.com/stevearc/oil.nvim) also see this discussion on Reddit (https://www.reddit.com/r/emacs/comments/1grqlf0/create_new_files_in_wdired/) . There is something that makes it so frictionless when the option is there to simply edit the filenames ...
